### PR TITLE
Scraper

### PR DESCRIPTION
**Pull request by kitorer**
* The db files are just for testing, we haven't made any decisions on hosting yet.
* Just pulling jsons from events.ucf

### CODE REVIEW

**Comment by rob-3** [comment]
Can you pull latest main and then `git rebase main` this? (Or use the "rebase branch" option on GitHub)

**Review by rob-3 — CHANGES_REQUESTED** [review]
Looks good overall, but the first backend PR raises some questions we should think about (won't block merge though).

* Do we store the entire JSON from other sources in another TEXT DB field? It'll be difficult to reconstruct that data later if we need it. I think we probably want to save it, if only for helping to deduplicate later on.
* How are we going to schedule scraping? I'm thinking that for now, a python script that runs the jobs on some schedule probably is simplest. Not sure how that will scale up to things like IG though

Good work, if you fix the blocking issues this can go in :)

**Comment by rob-3** [comment]
To update this you want the "update with rebase" option on GitHub, or
```bash
git switch main
git pull
git switch scraper
git rebase main
git push --force-with-lease
```

**Comment by rob-3** [comment]
Did you have issues getting the uuid() function calls in SQLite to work? It's perfectly ok to do it from python but the SQLite version should work if you have a recentish SQLite and it saves some work.

**Comment by kitorer** [comment]
yeah I did, but I'm also not on the most recent version of python rn. I can try that tho

**Comment by rob-3** [comment]
It's almost certainly an old SQLite version. Try installing the most recent build of SQLite if you want. Otherwise, the python solution is fine

File: repos/scrapers/UCF_events.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    cursor = conn.cursor()
+    cursor.execute('PRAGMA journal_mode=WAL')
+    cursor.execute('PRAGMA synchronous=NORMAL')
+
+    # Create a table to store the data
+    cursor.execute('''
+        CREATE TABLE IF NOT EXISTS events (
+            id BLOB PRIMARY KEY,
+            title TEXT,
+            description TEXT
```
Comment (rob-3) [inline repos/scrapers/UCF_events.py:22]: blocking: Missing comma

File: repos/scrapers/UCF_events.py [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+            url TEXT,
+            starts TEXT,
+            ends TEXT
+        ) STRICT 
+    ''')
+    
+    for event in data:
+        new_uuid = uuid.uuid4()
+        uuid_blob = new_uuid.bytes
+        cursor.execute('INSERT INTO events (id, title, description, url, starts, ends  ) VALUES (?, ?, ? ,? ,? ,?)',
```
Comment (rob-3) [inline repos/scrapers/UCF_events.py:32]: nit: random space after `ends`

**Review by rob-3 — CHANGES_REQUESTED** [review]

**Comment by rob-3** [comment]
If you rebase this onto main and force push, I will approve it!

**Review by rob-3 — APPROVED** [review]

**Pull request merged**